var/spack/repos/builtin/packages/py-cryptography/package.py

```diff
@@ -45,6 +45,8 @@ class PyCryptography(PythonPackage):
     depends_on('py-enum34',             type=('build', 'run'), when='^python@:3.4')
     depends_on('py-ipaddress',          type=('build', 'run'), when='^python@:3.3')
     depends_on('openssl@:1.0', when='@:1.8.1')
+    depends_on('openssl@1.0:', when='@2.0:')
+    depends_on('openssl@1.1:', when='@3.4:')
```

Comment:
Looking at the CHANGELOG, it seems like this should be:

* openssl 1.1.0+ for 3.2+
* openssl 1.0.2+ for 2.9+
* openssl 1.0.1+ for 1.7+
* openssl 1.0.0+ for 1.4+